### ISSUE TITLE
Policy Objects without displayName set cause error durring CSV Ownership file creation

### ISSUE THREAD

**Issue by cn4243** [Azure/enterprise-azure-policy-as-code#503]
Hi! 

EPAC 9.1.5
Powershell 7.3.7 and 7.4.0 tested.

I've ran into an issue where Export-AzPolicyResources will result in an error during the creation of the Ownership CSV file : 

```
===================================================================================================
Creating Ownership CSV file
===================================================================================================
Export-Csv: C:\Powershell\Modules\EnterprisePolicyAsCode\9.1.5\functions\Export-AzPolicyResources.ps1:840:12
Line |
 840 |  $allRows | Export-Csv -Path $ownershipCsvPath -NoTypeInformation -Enc …
     |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Object reference not set to an instance of an object.
```

I changed the code a bit to go row by row to identity which rows were causing trouble and pinned it down to displayName this way : 

```
kind                           Policy
owner                          unknownOwner
category                       Network
displayName
id                             /providers/Microsoft.Management/managementGroups/...
Export-Csv: C:\Powershell\Modules\EnterprisePolicyAsCode\9.1.5\functions\Export-AzPolicyResources.ps1:842:12
Line |
 842 |      $row | Export-Csv -Path $ownershipCsvPath -NoTypeInformation -App …
     |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Object reference not set to an instance of an object.
```

We unfortunately seem to have multiple custom policy definitions that do not have a displayName set, and only have the name property set.  The issue seems to be that that assigns a null value to displayName in the row object, rather than assigning an empty string object.  

I don't know what the intention was with using displayName, but a fix I found that works is if the displayName property is null, to use the name property : 

```
diff --git a/Scripts/Operations/Export-AzPolicyResources.ps1 b/Scripts/Operations/Export-AzPolicyResources.ps1
index fd2d665..ab95e53 100644
--- a/Scripts/Operations/Export-AzPolicyResources.ps1
+++ b/Scripts/Operations/Export-AzPolicyResources.ps1
@@ -413,7 +413,7 @@ foreach ($pacSelector in $globalSettings.pacEnvironmentSelectors) {
             else {
                 $rowObj.category = ""
             }
-            $rowObj.displayName = $properties.displayName
+            $rowObj.displayName = if ($properties.displayName) { $properties.displayName } else { $policyDefinition.name }   
             $rowObj.id = $policyDefinition.id
             $null = $allRows.Add($rowObj)
 
@@ -498,7 +498,7 @@ foreach ($pacSelector in $globalSettings.pacEnvironmentSelectors) {
             else {
                 $rowObj.category = ""
             }
-            $rowObj.displayName = $properties.displayName
+            $rowObj.displayName = if ($properties.displayName) { $properties.displayName } else { $policySetDefinition.name }
             $rowObj.id = $policySetDefinition.id
             $null = $allRows.Add($rowObj)
 
@@ -620,7 +620,7 @@ foreach ($pacSelector in $globalSettings.pacEnvironmentSelectors) {
             else {
                 $rowObj.category = ""
             }
-            $rowObj.displayName = $properties.displayName
+            $rowObj.displayName = if ($properties.displayName) { $properties.displayName } else { $policyAssignment.name }   
             $rowObj.id = $policyAssignment.id
             $null = $allRows.Add($rowObj)
 
@@ -776,7 +776,7 @@ foreach ($pacSelector in $globalSettings.pacEnvironmentSelectors) {
                     $rowObj.owner = $exemption.pacOwner
                 }
                 $rowObj.category = $exemption.exemptionCategory
-                $rowObj.displayName = $exemption.displayName
+                $rowObj.displayName = if ($exemption.displayName) { $exemption.displayName } else { $exemption.name }        
                 $rowObj.id = $exemption.id
                 $null = $allRows.Add($rowObj)
             }
```

**Comment by techlake** [comment]
We'll slot this into v10.0 (we are a little bit in a pickle here do to the big changes in V10.
ETA is about two weeks

**Comment by trichyz** [comment]
I had the same issue and the above changes fixed it. Thanks Marc!

**Comment by techlake** [comment]
Fixed in v10.0.0 released on 4/5/2024
